src/modm/driver/inertial/ixm42xxx.hpp

```diff
@@ -0,0 +1,196 @@
+// coding: utf-8
+// ----------------------------------------------------------------------------
+/*
+ * Copyright (c) 2023, Rasmus Kleist Hørlyck Sørensen
+ *
+ * This file is part of the modm project.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+// ----------------------------------------------------------------------------
+
+#ifndef MODM_IXM42XXX_HPP
+#define MODM_IXM42XXX_HPP
+
+#include "ixm42xxx_data.hpp"
+#include "ixm42xxx_definitions.hpp"
+#include "ixm42xxx_transport.hpp"
+
+namespace modm
+{
+
+/**
+ * @tparam	Transport	Either the I2C or SPI Transport Layer.
+ * @see Ixm42xxxTransportI2c
+ * @see Ixm42xxxTransportSpi
+ *
+ * @ingroup modm_driver_ixm42xxx
+ * @author	Rasmus Kleist Hørlyck Sørensen
+ */
+template < class Transport >
+class Ixm42xxx : public ixm42xxx, public Transport
+{
+private:
+    using Data = modm::ixm42xxxdata::Data;
+
+public:
+    /**
+     * @brief Constructor
+     *
+     * @param data A ixm42xxx::data object
+     * @param address the I2C address of the device
+     */
+    Ixm42xxx(Data &data, uint8_t address = 0b11010000);
+
+    /**
+     * @brief Initialize
+     *
+     * Sets used pins as output.
+     *
+     * The transport must be initialized by the user!
+     * @warning The SPI frequency must not exceed 2 MHz for this chip,
+     * or 1 MHz with burst mode.
```

Comment:
This is copy-paste from the ADIS16470 driver, isn't it :smiley:? None of the things in the comment applies for this device and driver.

Reply:
Ups yeah, this is unintentional! I have changed the comment for the initialize function.